kv/index_iter.go

```diff
@@ -246,3 +244,30 @@ func (c *kvIndex) SeekFirst(txn Transaction) (iter IndexIterator, err error) {
 	}
 	return &indexIter{it: it, idx: c, prefix: c.prefix}, nil
 }
+
+func (c *kvIndex) Exist(txn Transaction, indexedValues []interface{}, h int64) (bool, error) {
+	key, err := c.genIndexKey(indexedValues, h)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	value, err := txn.Get(key)
+	if IsErrNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	// For uniq index, the value of key is handle.
+	if c.unique {
+		handle, err := decodeHandle(value)
+		if err != nil {
+			return false, errors.Trace(err)
+		}
+
+		return h == handle, nil
```

Comment:
+1
For OnDuplicateKey insert, we need to get the duplicate row's key and update that row.